test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -420,35 +420,64 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 		})
 	})
 
-	// TODO: we have a dependency between tests(that in general bad practice, but saves us some tests run time),
-	// once we will want to run tests in the random order or without failFast we will need to refactor tests
 	Context("Updating of nodeSelector parameter and node labels", func() {
 		var mcp *machineconfigv1.MachineConfigPool
 		var newCnfNode *corev1.Node
 
 		newRole := "worker-test"
 		newLabel := fmt.Sprintf("%s/%s", testutils.LabelRole, newRole)
+		var mcpSelectorRemoved, labelsDeletion = false, false
 		newNodeSelector := map[string]string{newLabel: ""}
 
-		testutils.BeforeAll(func() {
+		existingMcpSelector := map[string]string{
```

Comment:
wouldn't it be better/safer to fetch the actual mcp selector - whatever it is and store it here?
is this possible?